Add OTLP log endpoint to convert log events to span annotations

making opened the pull request:
This is the first pull request that fixes gh-21.
More use case tests, optimizations and options will be added later.

How to check UI locally 

```bash
# Build the docker image
build-bin/docker/docker_build openzipkin-contrib/zipkin-otel:test
# Run the docker image
docker run --rm -p 9411:9411 docker.io/openzipkin-contrib/zipkin-otel:test
```

Run the example script using OpenAI+Python

```
cd collector-http/src/test/resources/python

python3 -m venv .venv
source .venv/bin/activate
export OTEL_EXPORTER_OTLP_ENDPOINT=http://localhost:9411
export OTEL_INSTRUMENTATION_GENAI_CAPTURE_MESSAGE_CONTENT=true
export OPENAI_API_KEY=sk-YOUR_API_KEY
./run.sh
cd -
```

Run the example script using OpenAI+NodeJs


```
cd collector-http/src/test/resources/nodejs

export OTEL_EXPORTER_OTLP_ENDPOINT=http://localhost:9411
export OTEL_INSTRUMENTATION_GENAI_CAPTURE_MESSAGE_CONTENT=true
export OPENAI_API_KEY=sk-YOUR_API_KEY
./run.sh
cd -
```


a screenshot and trace data is attached [here](https://github.com/openzipkin-contrib/zipkin-otel/issues/21#issuecomment-2557032138)

codefromthecrypt commented on collector-http/src/test/resources/nodejs/package-lock.json:
```diff
@@ -0,0 +1,3130 @@
+{
```
since this is just an example, maybe add .npmrc with "package-lock=false"?

making replied:
addressed in https://github.com/openzipkin-contrib/zipkin-otel/pull/26/commits/53648fa173e3124848092ece36296ef296b21fdf

codefromthecrypt commented on collector-http/src/test/java/zipkin2/collector/otel/http/ITLogEventEmission.java:
```diff
@@ -0,0 +1,421 @@
+    @Consumes("application/json")
+    @Produces("application/json")
+    public Map<String, Object> handleChatCompletion(String requestBody) {
+      Map<String, Object> response = new HashMap<>();
```
Not sure it works, but prefer `Map.of` stuff if it is usable. that or just a multi-line json literal. We can use a later JDK in tests than we do for the compiled binary (floor JRE version). This can be done later.

making replied:
Currently this project is compiled with Javav 8, probably just as it was when you scaffolded it.

Is there any reason to use this version? Would there be any problems with changing the base to Java 17?

codefromthecrypt reviewed (approved):
pretty cool to do real integration tests (with python and js), though maybe it could be less maintenance to send these signals directly in the integration test with otel java libs. 

I don't recall the current state of java and the otel logs api, but yeah pretty sure it can be done directly.

This sort of change is optional until maintaining the examples becomes troublesome. Also, we can drop one of the example languages if the other becomes a problem.

Aside: one configuration I would expect to come quickly is to use solely the "event.name" as the annotation value, so like probably via ENV.

Finally, seems like this is really getting to the point where we need a README, to describe how each signal is translated. e.g. traces (translated) metrics (dropped) logs (recorded as annotations if have a span context and event name)

making commented:
> Aside: one configuration I would expect to come quickly is to use solely the "event.name" as the annotation value, so like probably via ENV.

addressed in https://github.com/openzipkin-contrib/zipkin-otel/pull/26/commits/cb7a0dc99ed1d6ee67f45584fc38563029923c94

> I don't recall the current state of java and the otel logs api, but yeah pretty sure it can be done directly.

When I checked, it didn't seem to be supported in Java. If it's supported at the next time I review it, I'll switch to the Java SDK version.
At this time, I've added nodejs and python because I have very little knowledge of this feature and I wanted to test whether it works for your intended usecases.

codefromthecrypt commented on module/README.md:
```diff
@@ -50,6 +50,8 @@ for users that prefer a file based approach.
 |--------------------------------------|-------------------------------|----------------------------------------------------------|
 | `zipkin.collector.otel.http.enabled` | `COLLECTOR_HTTP_OTEL_ENABLED` | `false` disables the HTTP collector. Defaults to `true`. |
+| `zipkin.collector.otel.http.resource-attribute-prefix` | `COLLECTOR_OTEL_RESOURCE_ATTRIBUTE_PREFIX` | The prefix to use when converting otel resource attributes to span annotations. The default is to not prefix anything. |
+| `zipkin.collector.otel.http.log-event-name-attribute` | `COLLECTOR_OTEL_LOG_EVENT_NAME_ATTRIBUTE` | The otel attribute name that indicates whether to convert the upcoming Log Event to a Span. By default, `event.name` is used. |
```
I don't think it will ever be anything but event.name. Do we need a config around this? It might be more helpful to use config to decide what to do when it is `event.name`, like record the annotation with value == event.name attribute (search is usable) vs value == jsonification of the message (high cardinality so more cost, not usable in search)

making replied:
Ah, I seem to have misunderstood your comment.
The stability of the `event.name` attribute is "experimental", so I thought that might change in the future.
It's not a very useful configuration, so I'll remove it.

codefromthecrypt replied:
Most things in otel conventions seem to be experimental ;)

making replied:
Drop the configuration in https://github.com/openzipkin-contrib/zipkin-otel/pull/26/commits/004492a24567fc5af4dd6ee85884cfbe7a726d15

codefromthecrypt reviewed (approved):
:shipit:
